### PR TITLE
.

### PR DESCRIPTION
**Pull request by AriGute**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,11 +1,30 @@
...
+import { createBrowserRouter, RouterProvider } from 'react-router-dom';
+import TeamGenerator from './components/TeamGenerator';
+import About from './components/About';
+import Header from './components/Header';
 
 const App = () => {
-	return <div>{<Header />}</div>;
+	return (
+		<div className=' w-screen h-screen bg-blue-100 absolute top-0'>
+			<Header></Header>
```
Comment (Ronielo) [inline src/App.js:11]: All of the components you import do not need a closure tag

File: src/components/ImportPlayerList.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+				<textarea
+					className=' bg-gray-100 w-[300px] rounded'
+					name=''
+					id='playersToImport'
+					cols='30'
+					rows='5'
+					onChange={(e) => setPlayerList(e.target.value)}></textarea>
+				<button
+					className=' bg-gray-300 h-10 p-2 m-2 rounded w-[150px] hover:bg-gray-400'
+					onClick={() => {
```
Comment (Ronielo) [inline src/components/ImportPlayerList.js:25]: remove the onClick to a function
do not use that much logic inside the onClick without a function

File: src/components/PlayerCard.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const playerCard = ({ player, removeFunc }) => {
+	return (
+		<li className='border bg-gray-300 flex justify-between px-3 rounded'>
+			{player} {removeFunc && <button onClick={() => removeFunc(player)}>x</button>}
```
Comment (Ronielo) [inline src/components/PlayerCard.js:6]: create a new line, these have a `display: flex` the styling won't change but be more clear to read

File: src/components/TeamCompCard.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React, { useState } from 'react';
+import PlayerCard from './PlayerCard';
+
+const TeamCompCard = ({ list, name, removeFunc }) => {
+	return (
+		<div className=' text-center'>
+			<h1 className=' underline'>{name + ` [${list.length}]`}</h1>
+			<div className='w-[300px] min-h-[150px] bg-slate-100 rounded'>
+				<ul>
+					{list &&
```
Comment (Ronielo) [inline src/components/TeamCompCard.js:10]: i think you can remove it and just use `list?.map`

File: src/components/TeamGenerator.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+						[...playerList, ...importPlayers].forEach((player) => {
+							set.add(player);
+						});
+						setPlayerList([...set]);
+					},
+				}}></ImportPlayerList>
+			<TeamCompCard list={playerList} name={'Players'} removeFunc={removeAble}></TeamCompCard>
+			<div className='flex items-center rounded'>
+				<TeamCompCard list={teamA} name={'Team A'}></TeamCompCard>
+				<div className='flex flex-col'>
```
Comment (Ronielo) [inline src/components/TeamGenerator.js:29]: move this whole div to another component

File: src/components/TeamGenerator.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+					<button
+						className=' bg-green-300 h-10 p-2 m-2 rounded w-[150px] hover:bg-green-400'
+						onClick={function (params) {
```
Comment (Ronielo) [inline src/components/TeamGenerator.js:32]: create a function for this onClick

File: src/components/TeamGenerator.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+	const [teamA, setTeamA] = useState([]);
+	const [teamB, setTeamB] = useState([]);
+	const removeAble = (player) => {
+		setPlayerList(playerList.filter((p) => p != player));
+	};
+
+	return (
+		<div className='flex flex-col items-center'>
+			<ImportPlayerList
+				props={{
```
Comment (Ronielo) [inline src/components/TeamGenerator.js:17]: please remove props as prop and pass actual data

File: src/components/TeamGenerator.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+					callback: (importPlayers) => {
```
Comment (Ronielo) [inline src/components/TeamGenerator.js:18]: create the function outside the prop,
change the name of this callback to a proper name, for me i know its a callback no fucking clue what's the purpose

File: src/components/PlayerCard.js (hunk 3 above)
Comment (AriGute) [reply src/components/PlayerCard.js:6]: i didn't quite get it, can you give an example?

Comment (Ronielo) [reply src/components/PlayerCard.js:6]: Just get the button logic in the line below, don't put it in the same line too messy for the eyes

File: src/App.js (hunk 1 above)
Comment (AriGute) [reply src/App.js:11]: it's my emmet auto-complete. removing the closure tags everytime seems like a waste of time

Comment (Ronielo) [reply src/App.js:11]: Not for reading, it's a waste of lines and no need closures on every tag that doesn't contain anything inside of it. It's just for more clear code

**Pull request merged**